### ISSUE TITLE
NullPointerException warning while checking s3 bucket lifecycle configuration

### ISSUE THREAD

**Issue by eeshugerman** [spark-redshift-community/spark-redshift#46]
I see https://github.com/spark-redshift-community/spark-redshift/pull/30, but I think this _might_ be a different issue, because my lifecycle configuration does have a prefix.

It looks like so:
![image](https://user-images.githubusercontent.com/25518211/68172974-ae5a9680-ff36-11e9-9bf4-6520495a589d.png)

I should note there are a couple other prefixes in the bucket that do not have a lifecycle policy... not sure if that matters?

Feel free to close this if it is indeed addressed by #30.

```
19/11/05 01:21:28 WARN Utils$: An error occurred while trying to read the S3 bucket lifecycle configuration
java.lang.NullPointerException
        at java.lang.String.startsWith(String.java:1405)
        at java.lang.String.startsWith(String.java:1434)
        at io.github.spark_redshift_community.spark.redshift.Utils$$anonfun$5.apply(Utils.scala:140)
        at io.github.spark_redshift_community.spark.redshift.Utils$$anonfun$5.apply(Utils.scala:136)
        at scala.collection.Iterator$class.exists(Iterator.scala:917)
        at scala.collection.AbstractIterator.exists(Iterator.scala:1334)
        at scala.collection.IterableLike$class.exists(IterableLike.scala:77)
        at scala.collection.AbstractIterable.exists(Iterable.scala:54)
        at io.github.spark_redshift_community.spark.redshift.Utils$.checkThatBucketHasObjectLifecycleConfiguration(Utils.scala:136)
        at io.github.spark_redshift_community.spark.redshift.RedshiftWriter.saveToRedshift(RedshiftWriter.scala:390)
        at io.github.spark_redshift_community.spark.redshift.DefaultSource.createRelation(DefaultSource.scala:109)
        at org.apache.spark.sql.execution.datasources.SaveIntoDataSourceCommand.run(SaveIntoDataSourceCommand.scala:45)
        at org.apache.spark.sql.execution.command.ExecutedCommandExec.sideEffectResult$lzycompute(commands.scala:70)
        at org.apache.spark.sql.execution.command.ExecutedCommandExec.sideEffectResult(commands.scala:68)
        at org.apache.spark.sql.execution.command.ExecutedCommandExec.doExecute(commands.scala:86)
        at org.apache.spark.sql.execution.SparkPlan$$anonfun$execute$1.apply(SparkPlan.scala:131)
        at org.apache.spark.sql.execution.SparkPlan$$anonfun$execute$1.apply(SparkPlan.scala:127)
        at org.apache.spark.sql.execution.SparkPlan$$anonfun$executeQuery$1.apply(SparkPlan.scala:155)
        at org.apache.spark.rdd.RDDOperationScope$.withScope(RDDOperationScope.scala:151)
        at org.apache.spark.sql.execution.SparkPlan.executeQuery(SparkPlan.scala:152)
        at org.apache.spark.sql.execution.SparkPlan.execute(SparkPlan.scala:127)
        at org.apache.spark.sql.execution.QueryExecution.toRdd$lzycompute(QueryExecution.scala:80)
        at org.apache.spark.sql.execution.QueryExecution.toRdd(QueryExecution.scala:80)
        at org.apache.spark.sql.DataFrameWriter$$anonfun$runCommand$1.apply(DataFrameWriter.scala:676)
        at org.apache.spark.sql.DataFrameWriter$$anonfun$runCommand$1.apply(DataFrameWriter.scala:676)
        at org.apache.spark.sql.execution.SQLExecution$$anonfun$withNewExecutionId$1.apply(SQLExecution.scala:78)
        at org.apache.spark.sql.execution.SQLExecution$.withSQLConfPropagated(SQLExecution.scala:125)
        at org.apache.spark.sql.execution.SQLExecution$.withNewExecutionId(SQLExecution.scala:73)
        at org.apache.spark.sql.DataFrameWriter.runCommand(DataFrameWriter.scala:676)
        at org.apache.spark.sql.DataFrameWriter.saveToV1Source(DataFrameWriter.scala:285)
        at org.apache.spark.sql.DataFrameWriter.save(DataFrameWriter.scala:271)
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:498)
        at py4j.reflection.MethodInvoker.invoke(MethodInvoker.java:244)
        at py4j.reflection.ReflectionEngine.invoke(ReflectionEngine.java:357)
        at py4j.Gateway.invoke(Gateway.java:282)
        at py4j.commands.AbstractCommand.invokeMethod(AbstractCommand.java:132)
        at py4j.commands.CallCommand.execute(CallCommand.java:79)
        at py4j.GatewayConnection.run(GatewayConnection.java:238)
        at java.lang.Thread.run(Thread.java:748)
```

**Comment by mykolasmith** [comment]
Also seeing this issue...

**Comment by lucagiovagnoli** [comment]
@eeshugerman have you tried this in a bucket containing only prefixes with a lifecycle policy ? I'm trying to understand the steps to reproduce and the potential culprit.

**Comment by eeshugerman** [comment]
I have not, I'll report back when I have a chance to try that out.

Also, just to clarify the OP, I'm using `.option('tempdir', 's3n://mybucket/spark/data')`.

**Comment by julienbachmann** [comment]
see this PR #30 fix the issue but was not merged for now.

**Comment by eeshugerman** [comment]
I don't use the project any more so I can't verify, but looks like this has been fixed (#30 is now merged). Closing.
